source/fuzz/fact_manager/irrelevant_value_facts.cpp

```diff
@@ -60,13 +62,39 @@ bool IrrelevantValueFacts::PointeeValueIsIrrelevant(uint32_t pointer_id) const {
   return pointers_to_irrelevant_pointees_ids_.count(pointer_id) != 0;
 }
 
-bool IrrelevantValueFacts::IdIsIrrelevant(uint32_t pointer_id) const {
-  return irrelevant_ids_.count(pointer_id) != 0;
+bool IrrelevantValueFacts::IdIsIrrelevant(
+    uint32_t result_id, const DeadBlockFacts& dead_block_facts,
+    opt::IRContext* context) const {
+  // An id is irrelevant if it has been declared irrelevant.
+  if (irrelevant_ids_.count(result_id)) {
+    return true;
+  }
+
+  // An id is irrelevant if it is in a dead block.
+  return context->get_instr_block(result_id) &&
+         dead_block_facts.BlockIsDead(
+             context->get_instr_block(result_id)->id());
 }
 
-const std::unordered_set<uint32_t>& IrrelevantValueFacts::GetIrrelevantIds()
-    const {
-  return irrelevant_ids_;
+std::unordered_set<uint32_t> IrrelevantValueFacts::GetIrrelevantIds(
+    const DeadBlockFacts& dead_block_facts, opt::IRContext* context) const {
+  // Get all the ids that have been declared irrelevant.
+  auto irrelevant_ids = irrelevant_ids_;
+
+  // Get all the ids declared in dead blocks.
+  for (uint32_t block_id : dead_block_facts.GetDeadBlocks()) {
+    auto block = fuzzerutil::MaybeFindBlock(context, block_id);
+    if (!block) {
```

Comment:
I think it would be good to remark here that it's possible for the block to no longer exist in the module, e.g. if it has been merged with another block.

Reply:
Done